spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/ConfigFileApplicationListener.java

```diff
@@ -353,11 +348,14 @@ private void initializeProfiles() {
 			Binder binder = Binder.get(this.environment);
 			Set<Profile> activatedViaProperty = getProfiles(binder, ACTIVE_PROFILES_PROPERTY);
 			Set<Profile> includedViaProperty = getProfiles(binder, INCLUDE_PROFILES_PROPERTY);
-			List<Profile> otherActiveProfiles = getOtherActiveProfiles(activatedViaProperty, includedViaProperty);
+			Set<Profile> additionalProfiles = asProfileSet(
+					ConfigFileApplicationListener.this.additionalProfiles.toArray(new String[0]));
+			additionalProfiles.addAll(includedViaProperty);
+			List<Profile> otherActiveProfiles = getOtherActiveProfiles(activatedViaProperty, additionalProfiles);
 			this.profiles.addAll(otherActiveProfiles);
 			// Any pre-existing active profiles set via property sources (e.g.
 			// System properties) take precedence over those added in config files.
-			this.profiles.addAll(includedViaProperty);
+			this.profiles.addAll(additionalProfiles);
```

Comment:
To pass a parameter to the `getOtherActiveProfiles()` function, it seems to add `includedViaProperty`  to `additionalProfiles`, but this will cause the parameters of the function definition below to differ from passed parameters. 

Isn't it better to change the parameters of the 'getOtherActiveProfiles()' function?

```
private List<Profile> getOtherActiveProfiles(Set<Profile> activatedViaProperty,
				Set<Profile> includedViaProperty)
```

Reply:
Thanks for your comment, I will consider it.